src/videoConsumption.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * This module measures properties of video consumption.
+ * 
+ * @module WebScience.videoConsumption
+ */
+
+import * as debugging from "./debugging.js"
+import * as events from "./events.js"
+import * as matching from "./matching.js"
+import * as messaging from "./messaging.js"
+import * as pageManager from "./pageManager.js"
+import videoConsumptionContentScript from "./content-scripts/videoConsumption.content.js";
+
+/**
+ * @constant {debugging.debuggingLogger}
+ * @private
+ */
+const debugLog = debugging.getDebuggingLog("videoConsumption");
+
+/**
+ * Additional information about the video consumption event.
+ * @typedef {Object} VidoDataDetails
+ * @property {number} pageId - The ID for the page, unique across browsing sessions.
+ * @property {number} tabId - The ID for the tab containing the page, unique to the browsing session.
+ * @property {number} windowId - The ID for the window containing the page, unique to the browsing session.
+ * Note that tabs can subsequently move between windows.
+ * @property {string} url - The URL of the page loading in the tab, without any hash.
+ * @property {string} referrer - The referrer URL for the page loading in the tab, or `""` if
+ * there is no referrer.
+ * @property {number} pageVisitStartTime - The time when the page visit began.
+ * @property {number} pageVisitStopTime - The time when the underlying event fired.
+ * @property {number} playbackDuration - The total number of milliseconds videos on the page were playing.
+ * @property {string} ytChannel - The URL of the YouTube channel that uploaded the video (empty string if the page is not on YouTube).
+ * @property {boolean} privateWindow - Whether the page is in a private window.
```

Comment:
Would drop everything but the page ID, URL, playback duration, and YouTube channel. Leave the rest to other modules.